### PR TITLE
New tab for search

### PR DESCRIPTION
**Pull request by chStaiger**

### CODE REVIEW

File: ibridgesgui/search.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+        info = ""
+
+        # Get multiple paths from table
+        irods_paths = []
+        for idx in self.searchResTable.selectedIndexes():
+            irods_paths.append(IrodsPath(
+                                    self.session,
+                                    self.searchResTable.item(idx.row(), 1).text()))
+
+        select_dir = Path(QtWidgets.QFileDialog.getExistingDirectory(self, "Select Direcectory"))
```
Comment (qubixes) [inline ibridgesgui/search.py:138]:
```suggestion
        select_dir = Path(QtWidgets.QFileDialog.getExistingDirectory(self, "Select Directory"))
```

File: ibridgesgui/search.py [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            else:
+                coll = get_collection(self.session, result["COLL_NAME"])
+                table_data.append(('-C', coll.path, "",
+                                  coll.create_time.strftime("%d-%m-%Y"),
+                                  coll.modify_time.strftime("%d-%m-%Y")))
+            populate_table(self.searchResTable, len(table_data), table_data)
+
+    def download(self):
+        """Download selected data."""
+        info = ""
```
Comment (qubixes) [inline ibridgesgui/search.py:129]: This might need a comment. Does this override the info that you imported?

File: ibridgesgui/search.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+        if button_reply == QMessageBox.StandardButton.Yes:
+            overwrite = True
+            for ipath in irods_paths:
+                if ipath.exists():
+                    download(self.session, ipath, select_dir, overwrite=overwrite)
+                    self.logger.info("Downloading %s to %s, overwrite %s",
+                                str(ipath), select_dir, overwrite)
+                else:
+                    self.errorLabel.setText(f"Download failed. {str(ipath)} does not exist.")
+                    self.logger.info("Download failed: %s does nor exist", str(ipath))
```
Comment (qubixes) [inline ibridgesgui/search.py:163]:
```suggestion
                    self.logger.info("Download failed: %s does not exist", str(ipath))
```

File: ibridgesgui/search.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+        self.errorLabel.setText(f"Download finished, check {select_dir}")
```
Comment (qubixes) [inline ibridgesgui/search.py:164]:
```suggestion
        self.errorLabel.setText(f"Download finished, files put in {select_dir}")
```

**Review by qubixes — CHANGES_REQUESTED** [review]
One thing that I didn't get to work is to search for data objects. `%/bunny1.txt` does not return any results for me.

Of course, there are many possible improvements, but for now I think it is already very nice! I have mainly spelling errors this time.

**Comment by chStaiger** [comment]
> One thing that I didn't get to work is to search for data objects. `%/bunny1.txt` does not return any results for me.


Yes, that is a bug or a missing feature in the ibridges.search.search_data function ... Will be ammended in the next version of ibridges.

File: ibridgesgui/search.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+        if any([key.text() == "" and val.text() != "" for key, val in zip(self.keys, self.vals)]):
+            self.errorLabel.setText("There are metadata values without keys. Stop search.")
+            self.setCursor(QtGui.QCursor(QtCore.Qt.CursorShape.ArrowCursor))
+            return
+
+        if all([key.text() == "" for key in self.keys]):
+            key_vals = None
+        else:
+            # Replace empty values with the wild card, turn into search key_vals
+            key_vals = dict(zip([key.text() for key in self.keys],
```
[the remark this answers is not in the document]
Comment (chStaiger) [reply ibridgesgui/search.py:94]: Thank you. Yes this looks much better!

File: ibridgesgui/search.py (hunk 2 above)
Comment (chStaiger) [reply ibridgesgui/search.py]: It is actually not needed ... I removed it

**Pull request merged**